frontend/src/pages/Secret.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { user, logout } from '../reducers/user';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import styled from 'styled-components/macro';
+
+const URL = 'https://project-w20-auth-api.herokuapp.com/secrets';
+
+export const Secret = () => {
+  const dispatch = useDispatch();
+  const name = useSelector((store) => store.user.login.name);
+  const accessToken = useSelector((store) => store.user.login.accessToken);
+  const secretMessage = useSelector((store) => store.user.login.secretMessage);
+  const errorMessage = useSelector((store) => store.user.login.errorMessage);
+
+  const getSecret = () => {
+    fetch(URL, {
+      method: 'GET',
+      headers: { Authorization: accessToken },
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        } 
+        throw 'Could not get information. Make sure you are logged in and try again';
+      })
+      .then((json) => {
+        dispatch(
+          user.actions.setSecretMessage({ secretMessage: JSON.stringify(json) })
```

Comment:
Do you want it as a stringified object? Otherwise I think you could just write like this:
`user.actions.setSecretMessage({ json.secretMessage})`